src/influx_udp_worker.erl

```diff
@@ -146,61 +215,189 @@ to_key(N) when is_integer(N) ->
+  ?assertEqual(<<"test">>, to_tag_val("test")),
+  ?assertEqual(<<"test">>, to_tag_val(<<"test">>)),
+  ?assertEqual(<<"true">>, to_tag_val(true)),
+  ?assertEqual(<<"false">>, to_tag_val(false)),
+  ?assertEqual(<<"te\\ st">>, to_tag_val(<<"te st">>)),
+  ?assertEqual(<<"te\\,st">>, to_tag_val(<<"te,st">>)),
+  ?assertEqual(<<"te\\\"st">>, to_tag_val(<<"te\"st">>)),
+  ?assertEqual(<<"t\\ e\\\"s\\,t\\\"">>, to_tag_val(<<"t e\"s,t\"">>)),
+  ?assertEqual(<<"123">>, to_tag_val(123)).
+
+
+extract_time_test() ->
+  ?assertMatch({#{}, <<" 123">>}, extract_time(map, #{time => 123})),
+  ?assertMatch({#{}, <<" 123">>}, extract_time(map, #{"time" => 123})),
+  ?assertMatch({#{}, <<" 123">>}, extract_time(map, #{<<"time">> => 123})),
+  ?assertMatch({_, <<" 123">>}, extract_time(list, [{time, 123}])),
+  ?assertMatch({_, <<" 123">>}, extract_time(list, [{"time", 123}])),
+  ?assertMatch({_, <<" 123">>}, extract_time(list, [{<<"time">>, 123}])),
+  ?assertMatch({_, <<"">>}, extract_time(list, [{<<"tume">>, 123}])).
 
 encode_1_test() ->
-  Data = decode_data(encode(test, [ [{1, 1}], [{1, 3}] ])),
-  ?assertEqual(<<"test">>, get_series(Data)),
-  ?assertEqual([<<"1">>], get_columns(Data)),
-  ?assertEqual([[1], [3]], get_points(Data)).
+  ?assertEqual(
+    <<"test 1=1\ntest 1=3\n">>,
+    encode(test, [ [{1, 1}], [{1, 3}] ], [])
+  ).
 
 encode_2_test() ->
-  Data = decode_data(
-    encode("test", [ [{<<"1">>, 1}], [{<<"1">>, 3}] ])
-  ),
-  ?assertEqual(<<"test">>, get_series(Data)),
-  ?assertEqual([<<"1">>], get_columns(Data)),
-  ?assertEqual([[1], [3]], get_points(Data)).
+  ?assertEqual(
+    <<"test 1=1\ntest 1=3\n">>,
+    encode("test", [ [{<<"1">>, 1}], [{<<"1">>, 3}] ], [])
+  ).
 
 encode_3_test() ->
-  Data = decode_data(
-    encode(<<"test">>, [#{'1' => 1}, #{'1' => 3}])
-  ),
-  ?assertEqual(<<"test">>, get_series(Data)),
-  ?assertEqual([<<"1">>], get_columns(Data)),
-  ?assertEqual([[1], [3]], get_points(Data)).
+  ?assertEqual(
+    <<"test 1=1\ntest 1=3\n">>,
+    encode(<<"test">>, [#{'1' => 1}, #{'1' => 3}], [])
+  ).
 
 encode_4_test() ->
-  Data = decode_data(encode(<<"test">>, #{1 => 1})),
-  ?assertEqual(<<"test">>, get_series(Data)),
-  ?assertEqual([<<"1">>], get_columns(Data)),
-  ?assertEqual([[1]], get_points(Data)).
+  ?assertEqual(
+    <<"test 1=1\n">>,
+    encode(<<"test">>, #{1 => 1}, [])
+  ).
 
 encode_5_test() ->
-  ?assertNot(encode(<<"test">>, 'bla-bla')).
+  ?assertNot(encode(<<"test">>, 'bla-bla', [])).
+
+encode_with_tags_1_test() ->
+  ?assertEqual(
+    <<"test,host=eu-west,ip=1.1.1.1 1=1\n">>,
+    encode(<<"test">>, #{1 => 1}, [{host, "eu-west"}, {ip, '1.1.1.1'}])
+  ).
+
+encode_with_tags_2_test() ->
+  ?assertEqual(
+    <<"test,host=eu-west,ip=1.1.1.1 1=1\n">>,
+    encode(<<"test">>, #{1 => 1}, #{host => 'eu-west', ip => <<"1.1.1.1">>})
+  ).
+
+encode_with_tags_3_test() ->
+  ?assertEqual(
+    <<"test,host=eu-west,num=111 memory=\"high\",cpu=20\ntest,host=eu-west,num=111 memory=\"low\",cpu=30\n">>,
```

Comment:
Line 359 is too long:     <<"test,host=eu-west,num=111 memory=\"high\",cpu=20\ntest,host=eu-west,num=111 memory=\"low\",cpu=30\n">>,.